CRM/Core/ManagedEntities.php

```diff
@@ -464,6 +464,7 @@ protected function removeStaleEntity($dao) {
     // CRM_Core_BAO_Managed::on_hook_civicrm_post()
     if ($doDelete && CRM_Core_BAO_Managed::isApi4ManagedType($dao->entity_type)) {
       civicrm_api4($dao->entity_type, 'delete', [
+        'checkPermissions' => FALSE,
```

Comment:
(1) The earlier APIv3 variant did not need to set `check_permissions` because APIv3 defaults to `check_permissions=>FALSE`.

(2) The install/uninstall processes are system processes. While there should be some gatekeeping (*eg you can only enable or disable extensions if you have admin privileges*), the actual mechanics of the installation/uninstallation should not depend on *which user* performs the action.